File: logger/go/log.go
package main

import "fmt"

type MainFunc func(name string, x ...interface{})
type Func func(x ...interface{})

type Log struct {
	loggers	map[string]Func
	enabled	map[string]bool
	log	MainFunc
}

func New(f MainFunc) *Log {
	l := &Log{}
	l.loggers = make(map[string]Func)
	l.enabled = make(map[string]bool)
	l.log = f
	return l
}

func (l *Log) To(name string) Func {
	if lg, valid := l.loggers[name]; valid {
		return lg
	}
	lg := func(x ...interface{}) {
		if l.enabled[name] {
			l.log(name, x...)
		}
	}
	l.loggers[name] = lg
	l.enabled[name] = false
	return lg
}

func (l *Log) Enable(name string) {
	l.enabled[name] = true
}

func (l *Log) Disable(name string) {
	l.enabled[name] = false
}

func main() {
	log := New(func(name string, x ...interface{}) { 
		fmt.Print(name, ": ") 
		fmt.Println(x...)
	})
	log.To("alpha")("disabled")
	log.Enable("alpha")
	log.To("alpha")("enabled")

	lg := log.To("alpha")
	lg(1, 2, 3, 4)
	lg(1, 4)
	log.Disable("alpha")
	lg(3, 4, 5)
}

File: code/logger/js/log.js
function Log(){
    this.enabled = {};
};

Log.prototype.log = function(name, data){
    if( this.enabled[ name ] ){
        console.log( name, ": ", data );
    }
}

Log.prototype.create = function(name, enabled){
    var logger = this;
    if( typeof logger[name] !== "undefined" )
        throw "Cannot use name : " + name + ". Already in use.";

    logFun = function(){ logger.log( name, arguments ); };
    logFun.enable = function(){ logger.enable(name); };
    logFun.disable = function(){ logger.disable(name); };
    logger[name] = logFun;
    logger.enabled[name] = enabled || false;
};

Log.prototype.init = function(names, enabled){
    for(var i = 0; i < names.length; i += 1)
        this.create(names[i], enabled);
};

Log.prototype.enable = function(name){
    this.enabled[name] = true;
};

Log.prototype.disable = function(name){
    this.enabled[name] = false;
};

log = new Log();
log.init(["physics"])

log.physics.enable();
log.physics("1", 2, 3);
log.physics.disable();
log.physics("1", 2, 3);

// local alias
lg = log.physics;
lg("alpha", 1, 2, 3)